src/rust/engine/engine_pyo3/build.rs

```diff
@@ -31,15 +31,4 @@ fn main() {
   // NB: The native extension only works with the Python interpreter version it was built with
   // (e.g. Python 3.7 vs 3.8).
   println!("cargo:rerun-if-env-changed=PY");
-
-  if cfg!(target_os = "macos") {
-    // N.B. On OSX, we force weak linking by passing the param `-undefined dynamic_lookup` to
-    // the underlying linker. This avoids "missing symbol" errors for Python symbols
-    // (e.g. `_PyImport_ImportModule`) at build time when bundling the PyO3 sources.
-    // The missing symbols will instead by dynamically resolved in the address space of the parent
-    // binary (e.g. `python`) at runtime. We do this to avoid needing to link to libpython
-    // (which would constrain us to specific versions of Python).
-    println!("cargo:rustc-cdylib-link-arg=-undefined");
-    println!("cargo:rustc-cdylib-link-arg=dynamic_lookup");
-  }
```

Comment:
You might want to keep this bit around - unfortunately it looks like I'll have to revert this for PyO3 0.14.2: https://github.com/PyO3/pyo3/issues/1719

(I'll probably add something to the `pyo3-build-config` crate to do essentially what you've written here.)

Reply:
Hm, thanks! I'll pin to 14.1 for now. FYI I have an M1 machine also and I'm happy to help test things with that issue.